spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/ReorderMemory.scala

```diff
@@ -85,17 +90,45 @@ class ReorderMemory(inputFormat: MemoryData, outputFormat: MemoryData,
       }
     }
 
-    val fwdReorderPrimDesc = MklDnn.ReorderPrimitiveDescCreate(
-      realInput(0).getPrimitiveDescription(runtime),
-      realOutput(0).getPrimitiveDescription(runtime))
+    val needNotAttr = inputFormats()(0).dataType == DataType.F32 &&
+      outputFormats()(0).dataType == DataType.F32
+
+    val fwdReorderPrimDesc = if (needNotAttr) {
+      MklDnn.ReorderPrimitiveDescCreate(
+        realInput(0).getPrimitiveDescription(runtime),
+        realOutput(0).getPrimitiveDescription(runtime))
+    } else {
+      val attr = MklDnn.CreateAttr()
+      MklDnn.AttrSetIntOutputRoundMode(attr, 1)
+
+      if (_outputFormats(0).scales == null || !_outputFormats(0).scales.nonEmpty) {
+        if (_outputFormats(0).dataType == DataType.F32) {
+          _outputFormats(0).setScales(realInput(0).scales.map(1.0f / _))
+        } else {
+          _outputFormats(0).setScales(realInput(0).scales)
+        }
+        _outputFormats(0).setMask(realInput(0).mask)
+      }
+
+      require(_outputFormats(0).scales.nonEmpty)
+      require(_outputFormats(0).mask != -1)
+      MklDnn.AttrSetOutputScales(attr, _outputFormats(0).scales.length, _outputFormats(0).mask,
+        _outputFormats(0).scales)
+
+      MklDnn.ReorderPrimitiveDescCreateV2(
+        realInput(0).getPrimitiveDescription(runtime),
+        realOutput(0).getPrimitiveDescription(runtime),
+        attr)
+    }
+
     val fwdReorderPrim = MklDnn.PrimitiveCreate2(fwdReorderPrimDesc,
       Array(realInput(0).getPrimitive(runtime)), Array(0), 1,
       Array(realOutput(0).getPrimitive(runtime)), 1)
 
     updateOutputPrimitives = Array(fwdReorderPrim)
 
     // recover to original data
-    output = initTensor(realOutput(0))
+    output = initActivity(realOutput)
```

Comment:
why change to initActivity?

Reply:
It's the same. `initActivity` will call `initTensor`. I'll delete the change.